.github/workflows/e2e-test.yaml

```diff
@@ -0,0 +1,78 @@
+name: "E2E Test"
+on:
+  push:
+    branches:
+      - chore/e2e-test
+  workflow_dispatch:
+  schedule:
+    - cron: 0 22 * * *
+
+jobs:
+  e2e-test:
+    strategy:
+      fail-fast: false
+      matrix:
+        environment: [staging, production]
+        include:
+          - environment: staging
+            secret-name: AI_CORE_STAGING
+          - environment: production
+            secret-name: AI_CORE_PRODUCTION
+    name: "Build and Test"
+    runs-on: ubuntu-latest
+    env:
+      WGET_ARGS: "-qO- -S --content-on-error"
+      WGET_URL: "localhost:8080"
+    steps:
+      - name: "Checkout repository"
+        uses: actions/checkout@v4
+      - uses: pnpm/action-setup@v3
+        with:
+          version: 9.5.0
+      - uses: actions/setup-node@v4
+        with:
+          node-version: 22
+          cache: 'pnpm'
+      - run: pnpm i --frozen-lockfile
+      - name: Create .env file
+        env:
+          aicore: ${{ secrets[matrix.secret-name] }}
+        working-directory: ./sample-code
+        run: |
+          echo "aicore='$aicore'" > .env
+          url=$(echo "$aicore" | jq -r '.serviceurls.AI_API_URL' | sed 's|^https://||')
+          echo "Using AI Core ${{ matrix.environment }} instance on $url"
```

Comment:
we could even consider enforcing some pattern here to prevent copy/paste mistakes when updating the GitHub secrets..